docs/core-concepts/saturation-penalty.md

```diff
@@ -0,0 +1,80 @@
+---
+sidebar_position: 10
+---
+
+# Over-Saturation Penalty
+
+The penalty system in the saturation library is designed to incentivize proper risk management. It not only aims to fairly compensate LPs during various states of borrow and saturation utilization, but also prevents cascading liquidations. Therefore, a dynamic risk pricing system is required to create a self-balancing mechanism that maintains the protocol health.
+
+## Time-based penalty accrual
+
+The protocal calculates penalty accrual using a time based system. `accruePenalties` function in Saturation.sol takes a `duration` parameter that is defined as the `delta` between the current time and the last time penalty was accrued. See the call stack in `AmmalgamPair.sol`: `accrueSaturationPenaltiesAndInterest -> mintPenalties -> accruePenalties`.
+
+## Penalty threshold
+
+Penalties only starts accruing when a position saturation exceeds 85% (`START_SATURATION_PENALTY_RATIO_IN_MAG2 = 85`). When calculating the total saturation in penalty, see `calcTotalSatAfterLeafInclusive` in Saturation.sol, the threshold leaf is checked against the maxLeaf.
+
+When `threasholdLeaf > maxLeaf` it means that the leaf that corresponds to a penalty threshold of 85% saturation is higher than any leaf containing a position. Since `maxLeaf` is defined as `tree.highestSetLeaf`, this means that no position has been set to a leaf that is higher than the threshold leaf. In other words, no position has reached the 85% saturation threshold.
+
+On the other hand, if the a position has reached/surpassed the 85% saturation threshold, `maxLeaf > thresholdLeaf`, then the system loops through the leaves between `thresholdLeaf` and `maxLeaf` and add up all of the saturation (`satInLAssets`). The sum of all the saturation is then returned as the `satInLAssetsInPenalty`, which will be used to calculate the penalty rate.
+
+## Penalty rate calculation
+
+```
+Penalty formula:
+
+penaltyRate = ((1 - u₀) × f_interest(u₁) × totalDeposits) / (saturationInPenalty)
+
+ref: ` calcSaturationPenaltyRatePerSecondInWads` in Saturation.sol
```

Comment:
I find this as format a bit odd, also should be using latex to represent subscripts rather than using "u₀"
```suggestion
Penalty formula:
```math
penaltyRate = ((1 - u_0) × f_{interest}(u_1) × totalDeposits) / (saturationInPenalty)
\```
ref: ` calcSaturationPenaltyRatePerSecondInWads` in Saturation.sol
```

Remove the `\` its there to preserve the ``` in the suggestion.